### PR TITLE
Max three multiplication

### PR DESCRIPTION
**Pull request by ann-kolosova**
- Solution that finds max multiplication of three values in the list

### CODE REVIEW

File: Lesson08_Task01.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Output: Max value = "2016". Nums are: (-7, 12, -24)
+"""
+
+from itertools import permutations
+
+l1 = [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+
+copy_l1 = l1[:]
+
+# remove all 0s from copy_l1
```
Comment (antohaUa) [inline Lesson08_Task01.py:22]: Unfortunately solution is incorrect.
Please try you algorithm vs l1 = [-1, 0, 0, 0, 1, 2]  it gives Max value = "-2". Nums are: (-1, 1, 2)
But expected Max value = "0". Nums are: (0, 1, 2)

In addition for the next time please do NOT use itertools  )) I have shared with group solution already during Lesson_09 and group was notified that such solution we NOT use.
https://github.com/antohaUa/hillel_aqa_050623/blob/main/Lesson_09/00_prev_lesson.py#L28

Advice for the next time try to sort list and find multiplication max patterns. Or find your own algorithm )

**Review by antohaUa — CHANGES_REQUESTED** [review]

File: Lesson08_Task01.py (hunk 1 above)
Comment (ann-kolosova) [reply Lesson08_Task01.py]: fixed by using another solution

File: Lesson08_Task01.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+Input: [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+Output: Max value = "2016". Nums are: (-7, 12, -24)
+"""
+
+l1 = [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+
+copy_l1 = l1[:]
+
+# check if the length of the list contains at least 3 val
+if len(copy_l1) < 3:
```
Comment (antohaUa) [inline Lesson08_Task01.py:21]: Such blocks better orginize as single if with valueError inside. This is something new - but just for your information

```
if len(copy_l1) < 3:
    raise ValueError('At least 3 values should be in the input') 
```

And no else block required after this ^^

File: Lesson08_Task01.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    print('At least 3 values should be in the input')
+else:
+    max_mult = copy_l1[0] * copy_l1[1] * copy_l1[2]
+    max_nums = (copy_l1[0], copy_l1[1], copy_l1[2])
+
+    # find the maximum multiplication among three values
+    for i in range(len(copy_l1) - 2):
```
Comment (antohaUa) [inline Lesson08_Task01.py:28]: Straight solution with complexity O(n^3)  that is acceptable for junior level. But you see always such tasks have something hidden that you need to find.  In this task if we sort list
Maximum could be from two options
max1 = el[0] * [el1] * el[-1]    example: [-24, -7, -1, 0, 1, 2, 3, 5, 6, 12]
max2 = el[-3] * el[-2]*el[-1]    example: [-1, -1, 0, 1, 2, 3, 4, 5, 6, 12]

You can check other lists with different numbers - but max patterns will be the same. 
Then you just need to compare max1 vs max2 and choose correct one . Complexity in this case O(1)
Summary if you able to calculate complexity just keep in mind that iterview tasks in 99% could be solved with complexity up to O(n^2).

**Review by antohaUa — APPROVED** [review]

File: Lesson08_Task01.py (hunk 2 above)
Comment (ann-kolosova) [reply Lesson08_Task01.py:21]: created new file with this solution

File: Lesson08_Task01.py (hunk 3 above)
Comment (ann-kolosova) [reply Lesson08_Task01.py:28]: created new file with this solution

**Pull request merged**